kevm_pyk/src/kevm_pyk/utils.py

```diff
@@ -29,6 +30,12 @@
 from pyk.utils import hash_str
 
 
+def check_and_append_sentences(sents: List[KSentence], _sents: Iterable) -> None:
+    for sent in _sents:
+        assert isinstance(sent, KSentence)
+        sents.append(sent)
```

Comment:
As a Python-noob, this looks strange to me. Why do we need this dynamic type-check?

Is this any different from:

```python
check_and_append_sentences(sents: List[KSentence], _sents: Iterable[KSentence]) -> None:
    sents += list(_sents)
```

Reply:
It's because the things being passed in as `_sents` are often coming from functions that produce `List[KProduction]`, not `List[KSentence]`. But we have that `KProduction` is a subtype of `KSentence` (via subclassing), _but_ that _does not_ in Python mean that `List[KProduction]` is a subtype of `List[KSentence]`. So we have to actually iterate through the elements of the list and cast them each to the more general type explicitly. 

https://stackoverflow.com/questions/62814180/is-there-a-covariant-mutable-version-of-list and https://blog.daftcode.pl/covariance-contravariance-and-invariance-the-ultimate-python-guide-8fabc0c24278

Reply:
Thanks that is helpful.